### ISSUE TITLE
[Feature Request]  Option to disable sleep while bluetooth connection is active

### ISSUE THREAD

**Issue by aroum** [zmkfirmware/zmk#2198]
The ZMK checks the USB connection before starting the sleep timer. I would like to be able to check the BLE connections.

Now the keyboard falls asleep when I watch a video and I have to wake it up every time to pause the video.

I increased the time before falling asleep to an hour. Now, after turning off the PC, the keyboard still tries to connect to it for a long time.

I would like the keyboard to fall asleep 5 minutes after losing connection to the PC.

**Comment by petejohanson** [comment]
I think this is a really nice idea as a tweak or optional change to the current sleep code. Some folks may still prefer the current behavior, so probably work putting behavior a new Kconfig flag.

**Comment by newx-it** [comment]
Hello! I submitted a PR to solve this issue and also add another related flag/feature. Please let me know how it looks. 

[pull 3286](https://github.com/zmkfirmware/zmk/pull/3286)

**Comment by newx-it** [comment]
I've closed PR 3286 in favor of another PR with cleaned up commit history and a couple other fixes, It's available here: [PR 3288](https://github.com/zmkfirmware/zmk/pull/3288).

Apologies for the complication!
